components/blitz/src/omero/gateway/facility/BrowseFacility.java

```diff
@@ -603,12 +723,15 @@ public IObject findIObject(SecurityContext ctx, IObject o)
      * @param ids
      *            The ids of the datasets to load
      * @return A collection of {@link DatasetData}s
-     * @throws DSAccessException 
-     * @throws DSOutOfServiceException 
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
      */
     public Collection<DatasetData> getDatasets(SecurityContext ctx,
             long ownerId, Collection<Long> ids) throws DSOutOfServiceException, DSAccessException {
-        if (ids == null)
+        if (ids == null || ids.isEmpty() || ownerId < 0)
```

Comment:
Actually the current implementation doesn't make sense. `<0` as ownerId should mean the ownerId is ignored and the result should be the same like the according method without ownerId paramater is called, in that case `getDatasets(SecurityContext ctx, Collection<Long> ids)`. Returning an empty collection is probably not what one would expect. I'll fix that. Will also remove some code duplication.